mongo/tests/conftest.py

```diff
@@ -0,0 +1,136 @@
+# (C) Datadog, Inc. 2010-2017
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+import subprocess
+import os
+import logging
+import time
+
+import pymongo
+import pytest
+
+from datadog_checks.mongo import MongoDb
+
+from . import common
+
+log = logging.getLogger('conftest')
+
+
+@pytest.fixture
+def aggregator():
+    from datadog_checks.stubs import aggregator
+    aggregator.reset()
+    return aggregator
+
+
+@pytest.fixture
+def check():
+    check = MongoDb('mongo', {}, {})
+    return check
+
+
+@pytest.fixture(scope="session")
+def set_up_mongo():
+    cli = pymongo.mongo_client.MongoClient(
+        common.MONGODB_SERVER,
+        socketTimeoutMS=30000,
+        read_preference=pymongo.ReadPreference.PRIMARY_PREFERRED,)
+
+    foos = []
+    for _ in range(70):
+        foos.append({'1': []})
+        foos.append({'1': []})
+        foos.append({})
+
+    bars = []
+    for _ in range(50):
+        bars.append({'1': []})
+        bars.append({})
+
+    db = cli['test']
+    db.foo.insert_many(foos)
+    db.bar.insert_many(bars)
+
+    authDB = cli['authDB']
+    authDB.command("createUser", 'testUser', pwd='testPass', roles=[{'role': 'read', 'db': 'test'}])
+
+    db.command("createUser", 'testUser2', pwd='testPass2', roles=[{'role': 'read', 'db': 'test'}])
+
+    yield
+    tear_down_mongo()
+
+
+def tear_down_mongo():
+    cli = pymongo.mongo_client.MongoClient(
+        common.MONGODB_SERVER,
+        socketTimeoutMS=30000,
+        read_preference=pymongo.ReadPreference.PRIMARY_PREFERRED,)
+
+    db = cli['test']
+    db.drop_collection("foo")
+    db.drop_collection("bar")
+
+
+@pytest.fixture(scope="session")
+def spin_up_mongo():
+    """
+    Start a cluster with one master, one replica and one unhealthy replica and
+    stop it after the tests are done.
+    If there's any problem executing docker-compose, let the exception bubble
+    up.
+    """
+
+    env = os.environ
```

Comment:
The dev package also includes a way that may make this simpler as well. See here for an example - https://github.com/DataDog/integrations-core/pull/1870/files#diff-2b427522dd1eec04a03d07ae8e66525dR26 (`docker_run`)

Reply:
the new way errors out and I'm not gonna take the time to debug it after debugging this docker stuff for a week I'm done

Reply:
Fair, we should later invest some time to investigate this but shouldn't block merging this.